jaxrs/WFLY-15495_gRPC-support.adoc

```diff
@@ -0,0 +1,283 @@
+It can efficiently connect services implemented using various languages and frameworks.
+It is also applicable in the last mile of distributed computing to connect devices, mobile applications, and browsers to backend services.
+
+This RFE will add a gRPC subsystem to WildFly.
+It:
+
+* supports implementing gRPC services
+* supports consuming gRPC services
+* integrates with other WildFly components such as CDI, MP health, security or transaction
+* allows plain-text communication as well as TLS, and TLS with mutual authentication
+
+The gRPC subsystem provides a default configuration for all gRPC services and clients. The major part of the gRPC subsystem is a deployment unit processor which scans deployments for gRPC services, clients and interceptos. The default configuration can be overriden on a deployment basis.
+
+The gRPC subsystem defines a https://docs.wildfly.org/galleon/#_layers[Galleon layer] which will be responsible for providing the gRPC subsystem.
+
+=== Jakarta RPC Specification
+
+There's a Jakarta EE proposal for a specification called https://projects.eclipse.org/reviews/jakarta-rpc-creation-review[Jakarta RPC].
+This specification aims to make gRPC services and clients easier to implement in Java.
+The specification proposes a programming model for gRPC similar to JAX-RS and MicroProfile REST Client.
+As the time of writing, the proposal is not an official specification yet and very much wip.
+
+The scope of this RFE is *not* to implement Jakarta RPC.
+If it becomes a Jakarta EE specification at some point in time, there might be a follow-up RFE to support such a specification.
+Instead, this RFE is about to bring gRPC support to WildFly.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/WFLY-15495[WFLY-15495]
+* {issue-base-url}/EAP7-1819[EAP7-1819]
+
+=== Related Issues
+
+* {issue-base-url}/WFLY-13530[WFLY-13530]
+* {issue-base-url}/EAP7-1536[EAP7-1536]
+
+=== Dev Contacts
+
+* mailto:hpehl@redhat.com[Harald Pehl]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+== Requirements
+
+=== Hard Requirements
+
+* The gRPC subsystem provides a default configuration
+* The Galleon layer must provide the gRPC subsystem configuration.
+* gRPC services, clients and interceptors can be deployed independently of each other
+* gRPC services, clients and interceptors can be configured independently of each other
+* The gRPC subsystem integrates with the Elytron subsystem to provide TLS, and TLS with mutual authentication
+
+=== Nice-to-Have Requirements
+
+* WildFly quickstart for gRPC clients, services and interceptors
+* Support of https://github.com/grpc/grpc/blob/master/doc/health-checking.md[gRPC health check]
+* Support of https://github.com/grpc/grpc/blob/master/doc/server-reflection.md[gRPC relfection]
+* Support of https://smallrye.io/smallrye-mutiny/[Mutiny] API
+
```

Comment:
Nice to have: A gRPC chapter in the developer guide https://github.com/wildfly/wildfly/tree/main/docs/src/main/asciidoc/_developer-guide

Reply:
Good idea!